VisualPinball.Unity/VisualPinball.Unity/Game/CoilPlayer.cs

```diff
@@ -51,6 +59,8 @@ public void Awake(TableComponent tableComponent, IGamelogicEngine gamelogicEngin
 			_gamelogicEngine = gamelogicEngine;
 			_lampPlayer = lampPlayer;
 			_wirePlayer = wirePlayer;
+
+			_player = UnityEngine.Object.FindObjectOfType<Player>();

```

Comment:
I'd prefer passing it through the constructor like the other references, instead of globally fetching it. I know there are a few other places where we do that, but the problem with that approach is that if for whatever reason multiple player components need to be managed, this will break.

Reply:
Fixed! 

The only place left is inside the `DeviceCoil`. Not sure what to do here.

Reply:
Thanks! I'll have a look at `DeviceCoil`.

Reply:
Pushed an update. Passing it through the constructor as well, since everywhere it's instantiated we have access to the `Player` instance.

Reply:
awesome. Thank you!